src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/bukkit/listeners/GachaController.scala

```diff
@@ -0,0 +1,93 @@
+package com.github.unchama.seichiassist.subsystems.gacha.bukkit.listeners
+
+import cats.effect.{ConcurrentEffect, Sync}
+import cats.effect.ConcurrentEffect.ops.toAllConcurrentEffectOps
+import com.github.unchama.minecraft.actions.OnMinecraftServerThread
+import com.github.unchama.seichiassist.SeichiAssist
+import com.github.unchama.seichiassist.subsystems.gacha.bukkit.actions.BukkitDrawGacha
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaPrizesDataOperations
+import com.github.unchama.seichiassist.task.CoolDownTask
+import com.github.unchama.seichiassist.util._
+import com.github.unchama.util.syntax.Nullability.NullabilityExtensionReceiver
+import org.bukkit.ChatColor._
+import org.bukkit.event.block.Action
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.event.{EventHandler, Listener}
+import org.bukkit.inventory.EquipmentSlot
+import org.bukkit.{GameMode, Material}
+
+class GachaController[F[_]: ConcurrentEffect: Sync: OnMinecraftServerThread](
```

Comment:
```suggestion
class GachaController[F[_]: ConcurrentEffect: OnMinecraftServerThread](
```

`ConcurrentEffect`は自明に`Sync`です